### ISSUE TITLE
trial forces the creation of a new threadpool with max size 0 after running a TrialSuite

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3786]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#3786|
|Type|defect|
|Created|2009-04-17 16:38:49Z|
|Branch|https://github.com/twisted/twisted/tree/leave-threadpool-alone-3786|

`twisted.trial.runner.TrialSuite.run` overrides the base to call `_bail` after doing the normal `run` stuff.  The `_bail` method does various Twisted-related cleanup - specifically it fires the shutdown event and waits for all shutdown triggers to complete, and it resizes the reactor threadpool to 0.  This resize was introduced in r5030 in order to prevent trial from hanging at the end of the test suite waiting for non-daemon threads to exit (which never would by themselves).  This is not necessary now that the reactor cleans up its own threadpool, and it causes problems since by resizing the threadpool to 0, it forces the creation of a _new_ threadpool if the reactor has already cleaned its up.  A size 0 threadpool isn't useful for anything since it can't run any jobs.

<details><summary>Searchable metadata</summary>

```
trac-id__3786 3786
type__defect defect
reporter__exarkun exarkun
priority__high high
milestone__ 
branch__branches_leave_threadpool_alone_3786 branches/leave-threadpool-alone-3786
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__trial trial
keywords__ 
time__1239986329000000 1239986329000000
changetime__1240236025000000 1240236025000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [26732]) Branching to 'leave-threadpool-alone-3786'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [26733]) Stop calling resizeThreadPoolSize(0) at the end of a TestSuite run

refs [#3786](https://github.com/twisted/twisted/issues/3786)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun removed owner|
|-|-|

Did what I said.  [Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/leave-threadpool-alone-3786)

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_4"></a>|@jml set owner to @exarkun|
|-|-|

Looks good to me. Thanks for the clear description of the problem in the bug report. Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [26734]) Merge leave-threadpool-alone-3786

Author: exarkun
Reviewer: jml
Fixes: [#3786](https://github.com/twisted/twisted/issues/3786)

Remove the trial `TestSuite` cleanup code which resizes the reactor's thread
pool to 0.  The reactors take care of their own thread pool cleanup and this
resize is not only unnecessary, but it interfers with the process.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
